src/dockerfile/structure/dockerfile_parser.go

```diff
@@ -0,0 +1,274 @@
+package structure
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/bridgecrewio/yor/src/common"
+	"github.com/bridgecrewio/yor/src/common/logger"
+	"github.com/bridgecrewio/yor/src/common/structure"
+	"github.com/bridgecrewio/yor/src/common/tagging/tags"
+	"github.com/minamijoyo/tfschema/tfschema"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+type DockerfileParser struct {
+	Labels                 map[string]string
+	rootDir                string
+	taggableResourcesCache map[string]bool
+	providerToClientMap    sync.Map
+	dockerFileExists       bool
+	NewLabels              []TagItem
+}
+
+func (p *DockerfileParser) Name() string {
+	return "Dockerfile"
+}
+
+func (p *DockerfileParser) ValidFile(_ string) bool {
+	return true
+}
+
+func (p *DockerfileParser) ParseFile(filePath string) ([]structure.IBlock, error) {
+	var parsedBlocks []structure.IBlock
+	var block structure.Block
+	block.IsTaggable = false
+	sArray, errReadFile := readFile(filePath)
+	if errReadFile != nil {
+		log.Fatal(errReadFile)
+	}
+	for _, line := range sArray {
+		if strings.Contains(strings.ToLower(line), "label") {
+			tag := generateTag(line)
+			block.ExitingTags = append(block.ExitingTags, &tag)
+
+		}
+	}
+
+	files, err := ioutil.ReadDir(p.rootDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	foundConfig := false
+	for _, file := range files {
+		if filepath.Ext(file.Name()) == common.JSONFileType.Extension {
+			foundConfig = true
+			src, err := ioutil.ReadFile(file.Name())
+			if err != nil {
+				return nil, fmt.Errorf("failed to read file %s because %s", filePath, err)
+			}
+			var input InputTags
+			json.Unmarshal(src, &input.Input)
+			for key, value := range input.Input {
+				var tag tags.Tag
+				tag.Key = key
+				tag.SetValue(FormatString(value))
+				block.NewTags = append(block.NewTags, &tag)
+				block.IsTaggable = true
+
+			}
+		}
+	}
+	if !foundConfig {
+		return nil, fmt.Errorf("Failed to find input JSON file for configureation")
+	}
+
+	parsedBlocks = append(parsedBlocks, &block)
+
+	return parsedBlocks, nil
+
+}
+
+func FormatString(inputString string) string {
+	chars := []rune(inputString)
+	var newChars []rune
+	for i := 0; i < len(chars); i++ {
+		if string(chars[i]) != "\"" {
+			newChars = append(newChars, chars[i])
+		}
+	}
+	return string(newChars)
+
+}
+
+func (p *DockerfileParser) WriteFile(readFilePath string, blocks []structure.IBlock, writeFilePath string) error {
+	var tags []TagItem
+	for _, item := range blocks {
+		for _, oldTag := range item.GetExistingTags() {
```

Comment:
@sshintaku The logic of computing which tags need to be written is abstracted away inside the `block` struct. Please leverage it instead of writing a separate algorithm.

Reply:
I am trying to remediate this comment by trying to use the following algorithm to generate new tags:

var taggroup tagging.ITagGroup
taggroup.CreateTagsForBlock(block)

but since I don't know which properties need to be populated in the TagGroup object am kind of stuck.  Thanks.